### DISCUSSION TITLE
macos binaries for M1/M2 not being created

### DISCUSSION THREAD

**Discussion by eeholmes** [r-universe-org/help#241] [Q&A] [answered]
Thanks for opening up the r-universe platform! I am trying to create a r-universe for some packages with C++ code. And I am struggling a bit.

Here is the universe:
https://nmfs-opensci.r-universe.dev/builds

It did not build the M1/M2 binary for the atsar package:
https://nmfs-opensci.r-universe.dev/bin/macosx/big-sur-arm64/contrib/4.2/
https://nmfs-opensci.r-universe.dev/bin/macosx/x86_64-apple-darwin20/contrib/4.3/

Also when I look at any of the GitHub Actions it is not clear when the M1 build happens, though it clearly did for the other packages.
https://github.com/r-universe/nmfs-opensci/actions/runs/4780171485/jobs/8497906055

<img width="899" alt="image" src="https://user-images.githubusercontent.com/2545978/233986678-690a8705-6d52-4693-a9ae-6e999c72ade8.png">

Is there a way to get it to build the M1/M2 MacOs binaries?

**Comment by jeroen** [comment, marked as answer]
No not yet sadly. Ou runners are based on github actions, and those do not support M1 yet. They blogged recently that [apple-silicon-is-on-the-way](https://github.blog/2023-03-01-github-actions-introducing-faster-github-hosted-x64-macos-runners/#apple-silicon-is-on-the-way) so hopefully soon!

**Comment by eeholmes** [comment]
Ah ok. Why did the non-C++ packages get a "big-sur-arm64" binary created? Is it something about the C++ in the atsar package?

https://nmfs-opensci.r-universe.dev/bin/macosx/big-sur-arm64/contrib/4.2/

**Reply by jeroen** [reply]
Packages without compiled code are portable. We build them on intel, but they can be installed on either platform.
